src/plugin/dosdrv/xms.inc

```diff
@@ -101,7 +102,7 @@ HookHimem:
 	movw	$Int2f, %dx
 	int	$0x21
 	/* all done, UMB should work */
-	clc
```

Comment:
don't we need to clear CF still?

Reply:
`int $0x21` above should not fail, so why?

Reply:
I don't think int21/25 sets CF on error/success (not sure it can fail anyway?)

Reply:
But we need CF cleared here, not set.
Do you suspect it is set after 0x21?
Can you check if clc actually helps?

Reply:
> But we need CF cleared here, not set.

Yes

> Do you suspect it is set after 0x21?

I think it was set before the int21 and int21/25 doesn't set or clear it

> Can you check if clc actually helps?

Yes the umb driver is now loaded
~~~
Starting MS-DOS...
 
 
HIMEM is testing extended memory...done.
EMUFS host file and print access available
dosemu UMB driver installed
dosemu EMS driver rev 0.9 installed.
dosemu CDROM driver installed (V0.2)
emufs: redirector enabled
BLASTER=A220 I5 D1 H5 P330 T6
MIDI=SYNTH:2 MAP:E MODE:0
"Welcome to dosemu2 2.0pre8-20210717-3048-g461cd957a!"
Interactive Prompt!
C:\>
~~~

Reply:
OK, done, thanks.
This is an ms-dos bug then.

Reply:
> This is an ms-dos bug then.

I don't get that, int21/25 is a DOS v1 function. Those typically don't return error state via CF, but rely on setting a non-zero value in AL to indicate error. That being said, the int21/25 function can't fail, so doesn't even do that. See
http://www.ctyme.com/intr/rb-2602.htm

Reply:
OK but I don't think many other DOSes behave
the same way.